internal/store/store.go

```
@@ -619,6 +659,45 @@ func (s *Store) Delete(id string) error {
 }
 
+// DeleteChild removes a session only while it still hangs under parentID.
+// kill runs inside the same write transaction, which holds the database's
+// writer lock, so a placement cannot slip between the ownership check and
+// the delete, and a failed kill rolls the row back into place.
+func (s *Store) DeleteChild(id, parentID string, kill func() error) error {
+	tx, err := s.db.Begin()
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
```

golangci-lint (2.12.2)

[error] Error return value of `tx.Rollback` is not checked

(errcheck)